src/basic/env-util.c

```diff
@@ -779,7 +779,7 @@ int serialize_environment(FILE *f, char **environment) {
                 if (!ce)
                         return -ENOMEM;
 
-                fprintf(f, "env=%s\n", *e);
+                fprintf(f, "env=%s\n", ce);
```

Comment:
We don't allow variable values with newlines. I think it's something we should relax, but it's intentional (and documented), not a bug.